### ISSUE TITLE
Adding A/B testing capability to Focus

### ISSUE THREAD

**Issue by bbinto** [mozilla-mobile/focus-android#1030]
We shouldn't have any Firefox product anymore without an A/B testing framework.

Considering the nature of the app, I'd like to suggest adding Switchboard to Focus.

We could test things like #833 (different labeling) and the impact on usage of the feature.

**Comment by pocmo** [comment]
I recently talked to the Storage team (They maintain the server-side part for Switchboard) and for them it would be simple to add another experiment configuration.

As part of this work I'd like to rip the Switchboard code out of Fennec into its own library that we can use in all kinds of Android projects.

**Comment by bbinto** [comment]
Barbara to come up with some A/B tests

**Comment by bbinto** [comment]
One idea is to use switchboard for Insights surveys (ran by Matt's and Gemma's team)

Method: Invite existing users to participate in survey

Goal: 
- To gain insights about our existing Focus user, and to understand what features they consider important to, missing from Focus
- To gain insights about use cases when our users pick up Focus
- To understand what kind of users are most valuable to us (retain the best)
- Does convenience win over privacy?

More: https://docs.google.com/document/d/1lLG-DHLI9WZW2oRszjL7i0wVl96P06Y81lx1WjvKspM/edit#heading=h.nldib6sb6iko

**Comment by bbinto** [comment]
Eng: L
- re-factor Firefox switchboard code

**Comment by pocmo** [comment]
Storage/Operations bug to create a Kinto bucket/collection for storing the server side configuration:
https://bugzilla.mozilla.org/show_bug.cgi?id=1394515

**Comment by bbinto** [comment]
@bbinto to split up work to create first use case (UX required)

**Comment by Sdaswani** [comment]
- [https://github.com/mozilla-mobile/focus-android/issues/1431] Extract Switchboard Stuff from Fennec

**Comment by pocmo** [comment]
Depends on #1431 - I guess we should tackle the tasks in #1431 before this one? Although I filed an issue for integrating the library (#1581) - I guess this would replace this one here? Because after that we would have A/B testing capabilities in Focus.

**Comment by Sdaswani** [comment]
As I mentioned in https://github.com/mozilla-mobile/focus-android/issues/1431#issuecomment-340041757, it would be good to understand exactly what is meant by 'we would have A/B testing capabilities in Focus'. How do we set up A/B tests? How do we control them? What's the UI? Do we get reports? Etc.

**Comment by pocmo** [comment]
Closing this in favor of #1431.

> As I mentioned in #1431 (comment), it would be good to understand exactly what is meant by 'we would have A/B testing capabilities in Focus'. How do we set up A/B tests? How do we control them? What's the UI? Do we get reports? Etc.

I'll write a quick summary in the other issue.
